### PR TITLE
feat(tooltip): Create a `Tooltip` component

### PR DESCRIPTION
**Pull request by MadMed677**
# Description
PR creates a `Tooltip` component based on API specification provided [in issue](#1)

# API
```jsx
<Tooltip
  title={"Tooltip title"}
  position={TooltipPosition::Right}
  offset={4.0}
  arrow={true}
>
  <button>{"Hover on me"}</button>
</Tooltip>
```

### CODE REVIEW

File: src/components/tooltip.rs [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    Left,
+}
+
+#[derive(PartialEq, Properties)]
+pub struct TooltipProps {
+    /// Title of the tooltip
+    pub title: AttrValue,
+
+    /// Above which element the tooltip should be shown
+    pub children: Html,
```
[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs:30]: Oh, got it. I've made it by [official documentation](https://yew.rs/docs/concepts/function-components/children).
I'll change it. Thank you

File: src/components/tooltip.rs [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        // (!*visible).then_some(Some("hidden")),
+        (!*visible).then_some(Some("mm-invisible")),
+        (!*visible).then_some(Some("mm-opacity-0")),
+        (*visible).then_some(Some("mm-opacity-100")),
+    );
+
+    let common_arrow_classes = classes!(
+        "before:mm-content-['']",
+        "before:mm-absolute",
+        "before:mm-border-[8px]",
```
[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs:135]: Yep, agreed. But don't you know how to do it? I've change it into `format!("before:-mm-mt-[{TOOLTIP_ARROW_SIZE}px]"),` but it doesn't work. Even if I change `TOOLTIP_ARROW_SIZE` into `u8` there is still no class created by that name.
If I do print it I do have a correct naming `before:-mm-mt-[8px]` but I guess tailwind doesn't understand this construction and don't create a class for it

File: src/components/tooltip.rs [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+    /// Show arrow
+    ///  default value is `true`
+    #[prop_or(true)]
+    pub arrow: bool,
+
+    /// Offset in pixels
+    ///  default is 8px
+    #[prop_or(8.0)]
+    pub offset: f64,
```
[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs:49]: Yep, agreed. It's better to make it just `u16` for the public API. I'll change it

File: src/components/tooltip.rs [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    html! {
+        <span onmouseenter={on_mouse_enter} onmouseleave={on_mouse_leave} class={classes!(class.clone(), "mm-inline-block")}>
+            <div
+                class={final_classes}
+                style={format!("top: {top}px; left: {left}px;", top = coordinates.top, left = coordinates.left)}
+                ref={tooltip_ref}
+            >
+                { title.clone() }
+            </div>
+            { children.clone() }
```
[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs]: Done! e7aa5d52c8754272d451642a4670be5f7782ea5e

File: src/components/tooltip.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs]: Fixed! e7aa5d52c8754272d451642a4670be5f7782ea5e

File: src/components/tooltip.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs:135]: Oh, checked that it's not possible to add inline-styles for pseudo elements in our case it's `::before`. So there is no way to implement `2.` option.

So the third one is to keep it as it is :(

[the remark this answers is not in the document]
Comment (MadMed677) [reply src/components/tooltip.rs]: It has been done via CSS Variables, not it looks much better (from my POV :))
5897fd973e956ef3fcba2ecdffaff38d267be8e8

**Review by ri-nat — APPROVED** [review]
Now it's perfect!

**Pull request merged**